exchange/exchange.go

```diff
@@ -515,7 +519,12 @@ func (e *exchange) getAllBids(
 			reqInfo.PbsEntryPoint = bidderRequest.BidderLabels.RType
 			reqInfo.GlobalPrivacyControlHeader = globalPrivacyControlHeader
 
-			bids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, adjustmentFactor, conversions, &reqInfo, accountDebugAllowed, headerDebugAllowed)
+			//add call sign header for bidders where AdsCertDisable is enabled and request.ext.prebid.experiment.adscert.enabled is set to true
+			requestAdsCertEnabled := experiment != nil && experiment.AdsCert != nil && experiment.AdsCert.Enabled
+			bidderAdsCertEnabled := e.bidderInfo[string(bidderRequest.BidderName)].Experiment.AdsCert.Enable
+			addCallSignHeader := requestAdsCertEnabled && bidderAdsCertEnabled
+
+			bids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, adjustmentFactor, conversions, &reqInfo, e.adCertSigner, bidRequestOptions{accountDebugAllowed: accountDebugAllowed, headerDebugAllowed: headerDebugAllowed, addCallSignHeader: addCallSignHeader})
```

Comment:
I recommend moving up the declaration of `bidRequestOptions` to make it a bit more readable. This is a really long line. Something like:

```
requestAdsCertEnabled := experiment != nil && experiment.AdsCert != nil && experiment.AdsCert.Enabled
bidderAdsCertEnabled := e.bidderInfo[string(bidderRequest.BidderName)].Experiment.AdsCert.Enable
addCallSignHeader := requestAdsCertEnabled && bidderAdsCertEnabled

opts := bidRequestOptions{
	accountDebugAllowed: accountDebugAllowed,
	headerDebugAllowed:  headerDebugAllowed,
	addCallSignHeader:   addCallSignHeader,
}

bids, err := e.adapterMap[bidderRequest.BidderCoreName].requestBid(ctx, bidderRequest, adjustmentFactor, conversions, &reqInfo, e.adCertSigner, opts)

```

It also looks like there might be a chance to extract the logic for computing `addCallSignHeader` to a separate method.

Reply:
Modified